service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/managedIdentity/CreateAzureManagedIdentityStep.java

```diff
@@ -0,0 +1,110 @@
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import com.azure.core.management.Region;
+import com.azure.core.management.exception.ManagementException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Creates an Azure Managed Identity. Designed to run directly after {@link
+ * GetAzureManagedIdentityStep}.
+ */
+public class CreateAzureManagedIdentityStep implements Step {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(CreateAzureManagedIdentityStep.class);
+  private final AzureConfiguration azureConfig;
+  private final CrlService crlService;
+  private final ControlledAzureManagedIdentityResource resource;
+
+  public CreateAzureManagedIdentityStep(
+      AzureConfiguration azureConfig,
+      CrlService crlService,
+      ControlledAzureManagedIdentityResource resource) {
+    this.azureConfig = azureConfig;
+    this.crlService = crlService;
+    this.resource = resource;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    final AzureCloudContext azureCloudContext =
+        context
+            .getWorkingMap()
+            .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
+    var msiManager = crlService.getMsiManager(azureCloudContext, azureConfig);
+
+    try {
+      msiManager
+          .identities()
+          .define(resource.getManagedIdentityName())
+          .withRegion(resource.getRegion())
+          .withExistingResourceGroup(azureCloudContext.getAzureResourceGroupId())
+          .withTag("workspaceId", resource.getWorkspaceId().toString())
+          .withTag("resourceId", resource.getResourceId().toString())
+          .create(
+              Defaults.buildContext(
+                  CreateUserAssignedManagedIdentityRequestData.builder()
+                      .setName(resource.getManagedIdentityName())
+                      .setRegion(Region.fromName(resource.getRegion()))
+                      .setResourceGroupName(azureCloudContext.getAzureResourceGroupId())
+                      .setTenantId(azureCloudContext.getAzureTenantId())
+                      .setSubscriptionId(azureCloudContext.getAzureSubscriptionId())
+                      .build()));
+    } catch (ManagementException e) {
+      // Stairway steps may run multiple times, so we may already have created this resource. In all
+      // other cases, surface the exception and attempt to retry.
+      if (AzureManagementExceptionUtils.isExceptionCode(
+          e, AzureManagementExceptionUtils.CONFLICT)) {
+        logger.info(
+            "Azure Managed Identity {} in managed resource group {} already exists",
+            resource.getManagedIdentityName(),
+            azureCloudContext.getAzureResourceGroupId());
+        return StepResult.getStepResultSuccess();
+      }
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, e);
```

Comment:
Maybe use `return new StepResult(AzureManagementExceptionUtils.maybeRetryStatus(e), e)` .. think we do that in some places to not retry certain status codes (like 400).